src/pick_ik_plugin.cpp

```diff
@@ -239,15 +239,20 @@ class PickIKPlugin : public kinematics::KinematicsBase {
                 make_frame_tests(goal_frames,
                                  approximate_solution_position_threshold,
                                  approximate_solution_orientation_threshold);
-            auto const tip_frames = fk_fn(solution);
-            bool approx_solution_valid = true;
-            for (size_t i = 0; i < approx_frame_tests.size(); ++i) {
-                if (!approx_frame_tests[i](tip_frames[i])) {
-                    approx_solution_valid = false;
-                    break;
-                }
+
+            // If we have no cost threshold, we don't need to check the goals
+            if (params.approximate_solution_cost_threshold <= 0.0) {
+                goals.clear();
             }
```

Comment:
Instead of this, why don't we just set the value of `approx_solution_valid` to `true` directly, and only make/call the test function in the `else` case?

Reply:
We still need to call the `approx_solution_fn` function to check if the `approximate_solution_position_threshold` and `approximate_solution_orientation_threshold` are valid (if they are provided). It would be possible to use the lines of code that I deleted in my commit if no `approximate_solution_cost_threshold` is used. However, these are just the same lines of code that are also in the `approx_solution_fn`, which would mean code duplication.

Reply:
Understood! Sorry, I missed getting back to this in a timely manner.